src/code/z_map_disp.c

```diff
@@ -1,109 +1,1813 @@
+                        tempY = ((f32)D_801F53B0[i].pos.z - D_801BEBB8.sceneMidZ) * arg5 + ((arg4 / 2) + arg2);
+                        gSPTextureRectangle(POLY_OPA_DISP++, ((tempX - 1) << 2), ((tempY - 1) << 2), ((tempX + 1) << 2),
+                                            ((tempY + 1) << 2), G_TX_RENDERTILE, 0, 0, 0x0400, 0x0400);
+                    }
+                }
+            }
+        }
+
+        CLOSE_DISPS(play->state.gfxCtx);
+    }
+}
+
+#ifdef NON_MATCHING
+// https://decomp.me/scratch/nHYgx
+void func_80108558(PlayState* arg0, s32 arg1, s32 arg2, s32 arg3, s32 arg4, f32 arg5, s32 arg6) {
+    TransitionActorList* sp30;
+    // TransitionActorEntry* var_s0;
+    s32 temp_ft0;
+    s32 temp_ft0_2;
+    s32 var_s2;
+
+    sp30 = &D_801BEC24;
+    OPEN_DISPS(arg0->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80108124.s")
+    gDPPipeSync(POLY_OPA_DISP++);
+    gDPSetTextureLUT(POLY_OPA_DISP++, G_TT_NONE);
+    gDPSetRenderMode(POLY_OPA_DISP++, G_RM_AA_DEC_LINE, G_RM_NOOP2);
+    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, arg0->pauseCtx.alpha);
+    gDPSetEnvColor(POLY_OPA_DISP++, 0, 0, 0, 255);
+    gDPPipeSync(POLY_OPA_DISP++);
+    if (CHECK_DUNGEON_ITEM(DUNGEON_COMPASS, arg6)) {
+        gDPLoadTextureBlock_TEST(POLY_OPA_DISP++, &D_02002460, G_IM_FMT_IA, D_801BEAE0[1], D_801BEAF0[1], D_801BEB00[1],
+                                 D_801BEB10[1], D_801BEB20[1], D_801BEAD0[1], 8, 8, 0, G_TX_NOMIRROR | G_TX_WRAP,
+                                 G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+        for (var_s2 = 0; var_s2 < sp30->count; var_s2++) {
+            if ((func_80105328((s32)D_801F53B0[var_s2].params) != 0) &&
+                (func_80106BEC((s32)D_801BEBB8.unk42, (f32)D_801F53B0[var_s2].pos.y) != 0)) {
+                if (ABS_ALT(D_801F53B0[var_s2].id) != ACTOR_EN_HOLL) {
+                    temp_ft0 =
+                        ((((f32)D_801F53B0[var_s2].pos.x - D_801BEBB8.sceneMidX) * arg5) - 4) + ((arg3 / 2) + arg1);
+                    temp_ft0_2 =
+                        ((((f32)D_801F53B0[var_s2].pos.z - D_801BEBB8.sceneMidZ) * arg5) - 4) + ((arg4 / 2) + arg2);
+                    gSPTextureRectangle(POLY_OPA_DISP++, temp_ft0 << 2, temp_ft0_2 << 2, (temp_ft0 + 8) << 2,
+                                        (temp_ft0_2 + 8) << 2, G_TX_RENDERTILE, 0, 0, 0x0400, 0x0400);
+                }
+            }
+        }
+    }
 
+    CLOSE_DISPS(arg0->state.gfxCtx);
+}
+#else
+void func_80108558(PlayState*, s32, s32, s32, s32, f32, s32);
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80108558.s")
+#endif
+
+TexturePtr func_80108A10(s32 arg0) {
+    if ((arg0 >= 0) && (arg0 < 8)) {
+        return D_801BECE4[arg0];
+    }
+    if ((arg0 >= -8) && (arg0 < 0)) {
+        return D_801BED00[-arg0];
+    }
+    return &D_0C000000;
+}
+
+s32 func_80108A64(PlayState* play) {
```

Comment:
This function can also return true/false, and calls to it can be made boolean as well